### PR TITLE
 Redux and API

### PR DESCRIPTION
**Pull request by AsumVictor**
# Milestone 4: Redux and API
In this PR, I have added functionalities to fetch and display books data from external API

 ## Type of change
 * [ ] use redux createAsyncThunk  in `BookSlice.js` to handle async calls
 * [ ] Deleted hardcoded data for books

 # Checklist:
 * [ ]  My code follows the style guidelines of this project
 * [ ] I have performed a self-review of my code
 * [ ] There is no linter error in my code
 * [ ]  My changes generate no new warnings

### CODE REVIEW

File: src/redux/book/bookSlice.js [hunk 1]
```diff
@@ -1,51 +1,146 @@
...
-    type: 'remove',
-    payload: id,
-  });
-};
+// export const RemoveBook = (id) => (dispatch) => {
+//   dispatch({
+//     type: 'remove',
+//     payload: id,
+//   });
+// };
```
Comment (anagudelogu) [inline src/redux/book/bookSlice.js]: - [x] Please don't forget to remove old pieces of commented code, they make your code base look ugly and make it less readable. 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @AsumVictor, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/book/bookSlice.js (hunk 1 above)
Comment (AsumVictor) [reply src/redux/book/bookSlice.js]: Thank you very much for bringing that to be noticed!🤓

**Review by rotimiazeez — APPROVED** [review]
Hi @AsumVictor 👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Milestone requirements met 😉
- Clean codebase 👍🏼
- No linter error ✅️ 

# Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AsumVictor** [comment]
Thank you very much

**Pull request merged**